ntc_templates/templates/arista_eos_show_interfaces_status.textfsm

```diff
@@ -4,11 +4,11 @@ Value STATUS (connected|notconnect|inactive|errdisabled|disabled)
 Value VLAN_ID (\S+)
 Value DUPLEX (\S+)
 Value SPEED (\S+)
-Value TYPE (\S.*)
+Value TYPE (\S.*\S)
```

Comment:
This confused me a bit why have TYPE variable include a single non-whitespace at the end?

I would have thought TYPE would just be `\S+|Not Present`.

Reply:
> This confused me a bit why have TYPE variable include a single non-whitespace at the end?
> 
> I would have thought TYPE would just be `\S+|Not Present`.

A whole chunk of whitespace is captured otherwise.

I guess `Not Present` is the only two word value there ... I was keeping it loose in case there are other two word values.

I'm not hard set on what I have in `TYPE`. :grinning:

Reply:
I get pretty leery of that TYPE field since the .* is greedy and now we know there are two potential fields that can come after it (i.e. both `flags` and `encapsulation`). 

I think we should go with the or `Not Present` or else we very likely will capture the `Flags` or `encapsulation` field with the Type regex.

Reply:
> I get pretty leery of that TYPE field since the .* is greedy and now we know there are two potential fields that can come after it (i.e. both `flags` and `encapsulation`).

Totally understand the regex greediness concerns. :grinning:

> I think we should go with the or `Not Present` or else we very likely will capture the `Flags` or `encapsulation` field with the Type regex.

`\S+|Not Present` results in only the word `Not` being captured.

We could also opt for something like `\S+(\s+\S+)?` which is a bit longer but avoids the greediness concern ... though it only matches up to two words for `TYPE`.

Reply:
> Okay, good point--if we switch the order: `Not Present|\S+` then `Not Present` should be matched for the Type.

That can make it happier :grinning:

Will update this item.

> Wouldn't the `\S+(\s+\S+)?` still be problematic if either the Flags or Encapsulation fields actually contains data?

This isn't directed _at_ you - as long as we don't have output with one and both of those fields populated with real values we don't know what to filter for (in terms of values).

Those two fields (so far) seem minor given we don't have test data with values present.

If/when there's test data with values for one/both of those fields, I'd consider splitting off multiple States in the template as a long term solution.